resources/migrations/000_migrations.yaml

```diff
@@ -16096,6 +16096,59 @@ databaseChangeLog:
               column:
                 name: action_id
 
+  - changeSet:
+      id: v48.00-009
+      author: calherries
+      comment: Added 0.48.0 - Create table_privileges table
+      changes:
+        - createTable:
+            tableName: table_privileges
+            remarks: Table for user and role privileges by table
+            columns:
+              - column:
+                  autoIncrement: true
+                  constraints:
+                    nullable: false
+                    primaryKey: true
+                  name: id
+                  type: int
+                  remarks: 'row ID'
+              - column:
+                  name: table_id
+                  type: int
+                  remarks: 'Table ID'
+                  constraints:
+                    foreignKeyName: fk_table_privileges_table_id
+                    nullable: false
+                    referencedTableName: metabase_table
+                    referencedColumnNames: id
+                    deleteCascade: true
+              - column:
+                  name: is_current_user
```

Comment:
I find this column name confusing, is it to separate whether this is a role-based privilege?

if so how about `is_role_based_privileges`?

Reply:
Let me try to explain the purpose behind `is_current_user` and `role` first.

Without connection impersonation on, we can just use `is_current_user=true` to get the privileges for a table:
`(t2/select :model/TablePrivileges :is_current_user true :table_id table-id)`

With connection impersonation off, we have to use the user's `role`, which we'd get from their user attributes:
 `(t2/select :model/TablePrivileges :role role :table_id table-id)`

Of course, `is_current_user` is `true` for all rows in table_privileges right now, because we're only interested in those. But in the future, that might not be the case if an instance switched from having connection impersonation off to on and vice versa, or if we wanted to use connection impersonation for some users and no connection impersonation for others on the same table.

There's an argument to be made that we could just have NULL be a special value for `role`, meaning the privileges are for the current user. That would mean we could just have an index on `role` and `table_id`, as well as uniqueness constraint on those columns.

Reply:
I think I prefer having NULL be a special value of `role`, because we can enforce the invariant that there is only one row per table where the privileges are for the current user. We can a uniqueness constraint on the `role` and `table_id` column. You can't add a uniqueness constraint on the `is_current_user` and `table_id` column.

I'll remove the `is_current_user` column now.

Reply:
Done ca84ede3bb5b0a3533c9d093b3f8c99c4cfe83a5